frontend-v2/src/features/drug/Drug.tsx

```diff
@@ -227,81 +286,110 @@ const Drug: FC = () => {
-                        <Radio
-                          checked={isEfficacySelected(
+                    <div style={{ display: "flex", alignItems: "center" }}>
+                      <Radio
+                        checked={isEfficacySelected(
+                          efficacy_experiment as unknown as EfficacyRead,
+                        )}
+                        onClick={() =>
+                          handleSelectEfficacy(
                             efficacy_experiment as unknown as EfficacyRead,
-                          )}
-                          onClick={() =>
-                            handleSelectEfficacy(
-                              efficacy_experiment as unknown as EfficacyRead,
-                            )
-                          }
-                          disabled={isSharedWithMe}
-                        />
-                        <Typography color="primary">Select</Typography>
-                      </div>
-                    </Tooltip>
-                    <Tooltip title="Delete this efficacy-safety data">
-                      <IconButton
-                        onClick={() => setShowConfirmDelete(true)}
+                          )
+                        }
                         disabled={isSharedWithMe}
-                      >
-                        <DeleteIcon />
-                      </IconButton>
-                    </Tooltip>
-                    <ConfirmationDialog
-                      open={showConfirmDelete}
-                      title="Delete Efficacy-Safety Data"
-                      message="Are you sure you want to permanently delete this efficacy-safety data?"
-                      onConfirm={() => {
-                        deleteEfficacyExperiment(index);
-                        setShowConfirmDelete(false);
-                      }}
-                      onCancel={() => setShowConfirmDelete(false)}
+                      />
+                    </div>
+                  </Tooltip>
+                </TableCell>
+                <TableCell size="small">
+                  {isEditIndex === index ? (
+                    <TextField
+                      size="small"
+                      sx={{ flex: "1" }}
+                      label="Name"
+                      name={`efficacy_experiments.${index}.name`}
+                      control={control}
+                      textFieldProps={defaultProps}
                     />
-                  </div>
-                  <TextField
-                    sx={{ flex: "1" }}
-                    label="Name"
-                    name={`efficacy_experiments.${index}.name`}
-                    control={control}
-                    textFieldProps={defaultProps}
-                  />
-                  <Stack direction="row" spacing={2}>
+                  ) : (
+                    <Typography>
+                      {getValues(`efficacy_experiments.${index}.name`) || "-"}
+                    </Typography>
```

Comment:
I fixed the Projects listing so that you can select a project by clicking its name (see #604.) It might be worth making the same change here, marking the name up as the label for this row's radio button.

Reply:
Added